src/widgets-service.js

```diff
@@ -150,6 +151,16 @@ WidgetsService.prototype.createTimeSeriesModel = function (attrs, layer) {
   return widgetModel;
 };
 
+WidgetsService.prototype.setWidgetsState = function (state) {
```

Comment:
the deserilization should be done outside here, this function should not know where it's serialized (how would you do if the state comes from an ajax request instead of the url?)